nautobot/docs/development/core/local-k8s.md

```diff
@@ -0,0 +1,202 @@
+
+```text
+CONTAINER ID   IMAGE                           COMMAND                  CREATED          STATUS          PORTS                    NAMES
+...
+1f191e104357   local/nautobot-dev:local-py3.11 "watchmedo auto-rest…"   2 minutes ago    Up 37 seconds (healthy)   0.0.0.0:8081->8080/tcp, [::]:8081->8080/tcp nautobot-celery_worker-1
+d396ed398a4c   local/nautobot-dev:local-py3.11 "/docker-entrypoint.…"   2 minutes ago    Up 2 minutes (healthy)    0.0.0.0:8080->8080/tcp, :::8080->8080/tcp nautobot-nautobot-1
+...
+```
+
+Verify the Kind Docker network name is `kind`:
+
+```bash
+docker network ls
+```
+
+You should see all networks, including the `kind` network.
+
+```text
+NETWORK ID     NAME    DRIVER    SCOPE
+...
+6de64b53be75   kind    bridge    local
+...
+```
+
+Connect running Nautobot containers to the Kind network:
+
+```bash
+docker network connect kind nautobot-nautobot-1
+docker network connect kind nautobot-celery_worker-1
+```
+
+Now the selected Nautobot containers are connected to the Kind network and can access the Kind Kubernetes cluster.
+
+## Using `kubectl`
+
+To access the Kind Kubernetes cluster from the Nautobot container, you can use `kubectl` CLI tool.
+
+!!! Note
+    If you just want to use Python scripts to interact with the cluster, you can skip this chapter. However, it's recommended to install `kubectl` as it is a powerful tool for interacting with the Kubernetes cluster for debugging and testing.
+
+To verify the connection from Nautobot container to the Kind cluster:
+
+```bash
+# Open shell in the running Nautobot container
+invoke cli
+
+# Install kubectl
+apt update
+apt install kubernetes-client
+
+# Test the connection
+export KUBECONFIG=/source/development/kind-kube-config
+kubectl cluster-info
+```
+
+You should see the following output:
+
+```text
+Kubernetes control plane is running at https://nautobot-control-plane:6443
+CoreDNS is running at https://nautobot-control-plane:6443/api/v1/namespaces/kube-system/services/kube-dns:dns/proxy
+
+To further debug and diagnose cluster problems, use 'kubectl cluster-info dump'.
+```
+
+## Create a Testing Job
+
+You can verify the connection by creating the testing Kubernetes job:
+
+```bash
+invoke cli
+
+export KUBECONFIG=/source/development/kind-kube-config
+./development/create_k8s_job.py
+```
+
+!!! Note
+    For the following steps, the `kubectl` CLI tool must be installed as described in the previous chapter.
+
+To verify the job is running.
+
+```bash
+kubectl get job
+```
+
+You should see the following output:
+
+```text
+NAME       STATUS    COMPLETIONS   DURATION   AGE
+demo-job   Running   0/1           6s         6s
+```
+
+!!! Info
+    The job will run for 10 seconds and then complete.
+
+You can check the logs of the job:
+
+```bash
+kubectl logs job/demo-job --follow
+```
+
+Expected output:
+
+```text
+Hello, Kubernetes!
+```
+
+!!! Warning
+    You should delete the job after you are done with it, to be able to run the script again.
```

Comment:
What's considered best practice for k8s jobs in this regard? Always delete jobs after they run? Name repeated jobs with a UUID or an incrementing sequence number?

Reply:
There is `spec.ttlSecondsAfterFinished` job value to remove Job automatically after some time. Depends on the use case, for Nautobot Jobs to be run in K8s I would suggest to suffix `JobResult.id` to the job name.

Reply:
AWX adds the job ID to the name. Perhaps we should follow the same pattern.

As @snaselj mentioned, you can set `spec.ttlSecondsAfterFinished` for your job. I would set it to something low. Then, K8s will clean up the job after it is finished.